File: students/students/src/MON.Models/StudentModels/Class/StudentClassDualFormCompanyModel.cs
namespace MON.Models.StudentModels.Class
{
    using System;

    public class StudentClassDualFormCompanyModel
    {
        public int? Id { get; set; }
        public string Uic { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string Country { get; set; }
        public string District { get; set; }
        public string Municipality { get; set; }
        public string Settlement { get; set; }
        public string Area { get; set; }
        public string Address { get; set; }
        public string Url { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
    }
}

File: students/students/src/MON.SPA/src/models/studentClass/studentClassDualFormCompanyModel.js
import Constants from '@/common/constants.js';
import { uuid } from 'vue-uuid';

export class StudentClassDualFormCompanyModel {
  constructor(obj = {}, moment = {}) {
    this.id = obj.id;
    this.uid = uuid.v4();
    this.uic = obj.uic;
    this.name = obj.name;
    this.email = obj.email;
    this.phone = obj.phone;
    this.district = obj.district;
    this.municipality = obj.municipality;
    this.settlement = obj.settlement;
    this.area = obj.area;
    this.address = obj.address;
    this.url = obj.url;
    this.startDate = (obj.startDate && moment) ? moment(obj.startDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.endDate = (obj.endDate && moment) ? moment(obj.endDate).format(Constants.DATEPICKER_FORMAT) : '';
  }
}
